R/logging.R

```diff
@@ -63,6 +63,157 @@ message_wrap <-
     invisible(msg)
   }
 
+# issue cataloger --------------------------------------------------------------
+tune_env <-
+  rlang::new_environment(
+    data = list(progress_env = NULL, progress_active = FALSE, progress_catalog = NULL)
+  )
+
+# determines whether a supplied bar id references a tuning process that
+# uses the issue cataloger.
+uses_catalog <- function() {
+  isTRUE(tune_env$progress_active && !is_testing())
+}
+
+# determines whether a supplied bar id references an issue cataloger that
+# has not yet been terminated. this function is intended to guard calls to
+# `initialize_catalog()` inside of machinery that is wrapped by other tuning
+# method. e.g. `tune_bayes()` has an issue cataloger that ought not to be
+# overwritten when fitting over an initial grid.
+catalog_is_active <- function() {
+  tune_env$progress_active
+}
+
+# initializes machinery for the tune catalog inside of an environment.
+# the `env` should be an execution environment that persists throughout the
+# tuning process for a given tuning approach and exits once tuning is completed.
+#' @rdname tune-internal-functions
+#' @export
+initialize_catalog <- function(env = rlang::caller_env(), control) {
```

Comment:
I'd reverse the order of the arguments since one is optional and not normally specified

Reply:
You got it.